napari/utils/_appdirs.py

```diff
@@ -21,7 +26,7 @@
     appdirs.user_data_dir, _appname, _appauthor
 )
 user_config_dir: Callable[[], str] = partial(
-    appdirs.user_config_dir, _appname, _appauthor
+    appdirs.user_config_dir, _appname, _appauthor, version_string
```

Comment:
If I'm reading this right and looking above at line 13 plus this line: this will be added for every version, not just dev.
So this will result in `per-version` settings directory structures, like:
https://github.com/napari/napari/pull/5306
If that's to work for releases, then probably the setting reading/migration also needs to be implemented?